admin/class-insert-rule-data.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * Inserts rule data about a post to the database
+ *
+ * @package Accessibility_Checker
+ */
+
+namespace EDAC\Admin;
+
+/**
+ * Class for inserting rule data into the database
+ */
+class Insert_Rule_Data {
+
+	/**
+	 * Insert rule data into database
+	 *
+	 * @param object $post     The post object.
+	 * @param string $rule     The rule.
+	 * @param string $ruletype The rule type.
+	 * @param string $rule_obj The object.
+	 *
+	 * @return void|int
+	 */
+	public function insert( object $post, string $rule, string $ruletype, string $rule_obj ) {
+
+		if ( ! $post || ! $rule || ! $ruletype || ! $rule_obj ) {
+			return;
+		}
+
+		global $wpdb;
+		$table_name = $wpdb->prefix . 'accessibility_checker';
+
+		// set up rule data array.
+		$rule_data = array(
+			'postid'        => $post->ID,
+			'siteid'        => get_current_blog_id(),
+			'type'          => $post->post_type,
+			'rule'          => $rule,
+			'ruletype'      => $ruletype,
+			'object'        => esc_attr( $rule_obj ),
+			'recordcheck'   => 1,
+			'user'          => get_current_user_id(),
+			'ignre'         => 0,
+			'ignre_user'    => null,
+			'ignre_date'    => null,
+			'ignre_comment' => null,
+			'ignre_global'  => 0,
```

Comment:
Why are these `ignre` and not `ignore`?

Reply:
It's been four years but I believe I named it that way out of an abundance of caught to prevent conflicts or reserved keywords.

Reply:
That makes sense, I figured there was some reason behind it that wasn't just a simple typo haha

Hard to change it at this point if it was just a typo though :rofl: